opentelekomcloud/services/ces/resource_opentelekomcloud_ces_alarmrule.go

```diff
@@ -256,10 +232,11 @@ func ResourceAlarmRule() *schema.Resource {
 				Default:  true,
 			},
 			"alarm_action_enabled": {
-				Type:     schema.TypeBool,
-				Optional: true,
-				Default:  true,
-				ForceNew: true,
+				Type:         schema.TypeBool,
+				Optional:     true,
+				Default:      false,
+				ForceNew:     true,
+				AtLeastOneOf: []string{"alarm_actions", "ok_actions"},
```

Comment:
That's not a correct usage of the `AtLeastOneOf`. It refers to at least one of attributes in the list (or this attribute) to be set.

```
	// AtLeastOneOf is a set of attribute paths, including this attribute,
	// in which at least one of the attributes must be configured. This
	// implements the validation logic declaratively within the schema and can
	// trigger earlier in Terraform operations, rather than using create or
	// update logic which only triggers during apply.
	//
	// Only absolute attribute paths, ones starting with top level attribute
	// names, are supported. Attribute paths cannot be accurately declared
	// for TypeList (if MaxItems is greater than 1), TypeMap, or TypeSet
	// attributes. To reference an attribute under a single configuration block
	// (TypeList with Elem of *Resource and MaxItems of 1), the syntax is
	// "parent_block_name.0.child_attribute_name".
	AtLeastOneOf []string
```

Reply:
I see, what would you suggest to implement a state like this?
"If you set alarm_action_enabled to true, you must specify either alarm_actions or ok_actions."

Reply:
I guess `requiredWith` on `alarm_actions`/ `ok_actions` would fit here

Reply:
Nope it will make `alarm_action_enabled` mandatory if `alarm_actions`/`ok_actions` are enabled

Reply:
where is the AtLeastOneOf from?

Reply:
Schema struct

Reply:
Sorry I didn't find it... could you give me a link?

Reply:
> I see, what would you suggest to implement a state like this? "If you set alarm_action_enabled to true, you must specify either alarm_actions or ok_actions."

This should be implemented inside either in `CustomizeDiff` schema attribute (`schema.CustomizeDiffFunc`), there are some examples of that, or in exact update/create operation.


> Sorry I didn't find it... could you give me a link?

You can check the schema documentation:
https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/schema.go

Reply:
updated